GetStream/stream-video-flutter
# feat: review missing slots

### 🎯 Goal

Closes https://github.com/GetStream/stream-video-flutter/issues/220

Add the missing content slots (widget builders).

### 🛠 Implementation details

Tried to make all the components that we ship as standalone as possible. The following code snippet should produce the same result as `StreamCallContainer(call)`: 

```
StreamCallContainer(
    call: call,
    callConnectOptions: options,
    incomingCallBuilder: (context, call, callState) {
        return IncomingCallContent(
            call: call,
            callState: callState,
        );
    },
    outgoingCallBuilder: (context, call, callState) {
        return OutgoingCallContent(
            call: call,
            callState: callState,
        );
    },
    callContentBuilder: (context, call, callState) {
        return StreamCallContent(
            call: call,
            callState: callState,
            callAppBarBuilder: (context, call, callState) {
                return CallAppBar(call: call);
            },
            callParticipantsBuilder: (context, call, callState) {
                return StreamCallParticipants(
                    call: call,
                    participants: callState.callParticipants,
                    callParticipantBuilder: (context, call, participant) {
                        return StreamCallParticipant(
                            call: call,
                            participant: participant,
                        );
                    },
                    localVideoBuilder: (context, call, participant, localVideoChild) {
                        return StreamLocalVideo(
                            call: call,
                            localParticipant: participant,
                            child: localVideoChild,
                        );
                    },
                    screenShareContentBuilder: (context, call, participant) {
                        return ScreenShareContent(
                            call: call,
                            participant: participant,
                        );
                    },
                    screenShareParticipantBuilder: (context, call, participant) {
                        return ScreeShareParticipant(
                            call: call,
                            participant: participant,
                        );
                    },
                );
            },
            callControlsBuilder: (context, call, callState) {
                return StreamCallControls.withDefaultOptions(
                    call: call,
                    localParticipant: callState.localParticipant!,
                );
            },
            participantsInfoBuilder: (context, call) {
                final usersProvider = StreamUsersConfiguration.of(context);
                return StreamCallParticipantsInfoMenu(
                    call: call,
                    usersProvider: usersProvider,
                );
            },
        );
    },
);
```

### 🧪 Testing

1. Run the example app
2. Apply the patch below
3. Verify that the app works as before

<details>

<summary>Provide the patch summary here</summary>

```
Index: packages/stream_video_flutter/example/lib/screen/home_screen.dart
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/packages/stream_video_flutter/example/lib/screen/home_screen.dart b/packages/stream_video_flutter/example/lib/screen/home_screen.dart
--- a/packages/stream_video_flutter/example/lib/screen/home_screen.dart	(revision 3174a6af3ed8f0d2659724363bf3a825c921d2ce)
+++ b/packages/stream_video_flutter/example/lib/screen/home_screen.dart	(date 1677631802923)
@@ -86,6 +86,73 @@
           child: StreamCallContainer(
             call: call,
             callConnectOptions: options,
+            incomingCallBuilder: (context, call, callState) {
+              return IncomingCallContent(
+                call: call,
+                callState: callState,
+              );
+            },
+            outgoingCallBuilder: (context, call, callState) {
+              return OutgoingCallContent(
+                call: call,
+                callState: callState,
+              );
+            },
+            callContentBuilder: (context, call, callState) {
+              return StreamCallContent(
+                call: call,
+                callState: callState,
+                callAppBarBuilder: (context, call, callState) {
+                  return CallAppBar(call: call);
+                },
+                callParticipantsBuilder: (context, call, callState) {
+                  return StreamCallParticipants(
+                    call: call,
+                    participants: callState.callParticipants,
+                    callParticipantBuilder: (context, call, participant) {
+                      return StreamCallParticipant(
+                        call: call,
+                        participant: participant,
+                      );
+                    },
+                    localVideoBuilder:
+                        (context, call, participant, localVideoChild) {
+                      return StreamLocalVideo(
+                        call: call,
+                        localParticipant: participant,
+                        child: localVideoChild,
+                      );
+                    },
+                    screenShareContentBuilder: (context, call, participant) {
+                      return ScreenShareContent(
+                        call: call,
+                        participant: participant,
+                      );
+                    },
+                    screenShareParticipantBuilder:
+                        (context, call, participant) {
+                      return ScreeShareParticipant(
+                        call: call,
+                        participant: participant,
+                      );
+                    },
+                  );
+                },
+                callControlsBuilder: (context, call, callState) {
+                  return StreamCallControls.withDefaultOptions(
+                    call: call,
+                    localParticipant: callState.localParticipant!,
+                  );
+                },
+                participantsInfoBuilder: (context, call) {
+                  final usersProvider = StreamUsersConfiguration.of(context);
+                  return StreamCallParticipantsInfoMenu(
+                    call: call,
+                    usersProvider: usersProvider,
+                  );
+                },
+              );
+            },
           ),
         ),
       ),

```

</details>


### ☑️Contributor Checklist

#### General

### ☑️Reviewer Checklist
